xgi/linalg/matrix.py

```diff
@@ -59,48 +59,41 @@ def incidence_matrix(
     node_dict = dict(zip(node_ids, range(num_nodes)))
     edge_dict = dict(zip(edge_ids, range(num_edges)))
 
-    if node_dict and edge_dict:
```

Comment:
As far as I can tell, the main change in this block is the deletion of this `if` statement. Could you please explain why it is no longer necessary?

Reply:
Every test that I wrote couldn't access that part of the code because when the hypergraph was empty, it would only hit the if-statement at the beginning. This moves all the handling of empty hypergraphs to the beginning and I've changed the beginning if-statement to now handle no nodes as well (before it just handled no edges)

Reply:
That makes sense, thanks.